### PR TITLE
Fix the repos_file_branch macro and multiversion wf

### PR DESCRIPTION
**Pull request by christophfroehlich**
- otherwise the generated links were wrong in the jazzy version
- multiversion workflow failed for PRs, because the base branch is not called "rolling" then

### CODE REVIEW

File: make_help_scripts/fix_index.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    # Copy the contents of the base_branch directory to the master directory
+    shutil.copytree(os.path.join(builddir, 'html', base_branch), os.path.join(builddir, 'html', 'master'), dirs_exist_ok=True)
+
+    # Patch the index.html file in the master directory
+    with open(os.path.join(builddir, 'html', 'master', 'index.html'), 'w') as f:
+        f.write(f'<html><head><meta http-equiv="refresh" content="0; url=../{base_branch}/index.html" /></head></html>')
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Fix index.')
+    parser.add_argument('--builddir', required=True, help='Build directory.')
```
Comment (saikishor) [inline make_help_scripts/fix_index.py]:
```suggestion
    parser.add_argument('--build-dir', required=True, help='Build directory.')
```

File: Makefile [hunk 2]
```diff
@@ -113,10 +107,7 @@ multiversion-with-api: Makefile
 	@echo Step 5: Building multiversion API
 	./make_help_scripts/create_api_multi_version.py
 	@echo Step 6: Create correct index + legacy master version
-	@echo "<html><head><meta http-equiv=\"refresh\" content=\"0; url=rolling/index.html\" /></head></html>" > "$(BUILDDIR)"/html/index.html
-# legacy, renamed Rolling version from "master" to "rolling"
-	@cp -r "$(BUILDDIR)"/html/rolling/ "$(BUILDDIR)"/html/master
-	@echo "<html><head><meta http-equiv=\"refresh\" content=\"0; url=../rolling/index.html\" /></head></html>" > "$(BUILDDIR)"/html/master/index.html
+	./make_help_scripts/fix_index.py --builddir $(BUILDDIR)
```
Comment (saikishor) [inline Makefile]:
```suggestion
	./make_help_scripts/fix_index.py --build-dir $(BUILDDIR)
```

File: Makefile [hunk 3]
```diff
@@ -95,10 +92,7 @@ multiversion-with-errors: Makefile
 	@echo Step 3: Deleting temporary commits
 	./make_help_scripts/delete_tmp_commits.py
 	@echo Step 4: Create correct index + legacy master version
-	@echo "<html><head><meta http-equiv=\"refresh\" content=\"0; url=rolling/index.html\" /></head></html>" > "$(BUILDDIR)"/html/index.html
-# legacy, renamed Rolling version from "master" to "rolling"
-	@cp -r "$(BUILDDIR)"/html/rolling/ "$(BUILDDIR)"/html/master
-	@echo "<html><head><meta http-equiv=\"refresh\" content=\"0; url=../rolling/index.html\" /></head></html>" > "$(BUILDDIR)"/html/master/index.html
+	./make_help_scripts/fix_index.py --builddir $(BUILDDIR)
```
Comment (saikishor) [inline Makefile]:
```suggestion
	./make_help_scripts/fix_index.py --build-dir $(BUILDDIR)
```

File: Makefile [hunk 4]
```diff
@@ -81,10 +81,7 @@ multiversion: Makefile
...
```
Comment (saikishor) [inline Makefile]:
```suggestion
	./make_help_scripts/fix_index.py --build-dir $(BUILDDIR)
```

File: make_help_scripts/fix_index.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    args = parser.parse_args()
+    fix_index(deploy_defines.base_branch, args.builddir)
```
Comment (saikishor) [inline make_help_scripts/fix_index.py]:
```suggestion
    fix_index(deploy_defines.base_branch, args.build_dir)
```

**Review by saikishor — CHANGES_REQUESTED** [review]
Just a nitpicking change. Rest looks good to me. I recommend checking if these proposed changes work.

**Comment by christophfroehlich** [comment]
@saikishor CI is happy

**Review by saikishor — APPROVED** [review]
LGTM

**Pull request merged**